vendor/bat-native-ads/src/bat/ads/internal/logging.h

```diff
@@ -1,26 +1,70 @@
-/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
  * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #ifndef BAT_ADS_INTERNAL_LOGGING_H_
 #define BAT_ADS_INTERNAL_LOGGING_H_
 
-#include "bat/ads/ads_client.h"
+#include <ostream>
 
-#define ADS_LOG_INFO \
-  ads_client_->Log(__FILE__, __LINE__, ::ads::LogLevel::LOG_INFO)
+namespace ads {
 
-#define ADS_LOG_WARNING \
-  ads_client_->Log(__FILE__, __LINE__, ::ads::LogLevel::LOG_WARNING)
+// Undefine base::logging macros so that Brave Ads can use LOG, LOG_IF, VLOG and
+// VLOG_IF
 
-#define ADS_LOG_ERROR \
-  ads_client_->Log(__FILE__, __LINE__, ::ads::LogLevel::LOG_ERROR)
+#undef LOG
```

Comment:
should we first check if it's defined before we undefine it?

Reply:
`#undef` is ignored if the specified identifier is not currently defined as a macro name